decisions/version-compatibility.md

```diff
@@ -0,0 +1,131 @@
+|            | 1.02 | not compatible | **compatible** | **compatible**                    | **compatible**                    | not compatible |
+|            | 1.06 | not compatible | not compatible | **compatible**                    | **compatible**                    | not compatible |
+|            | 1.07 | not compatible | not compatible | ~~**compatible**~~ not compatible | **compatible**                    | not compatible |
+|            | 2    | not compatible | not compatible | not compatible                    | ~~**compatible**~~ not compatible | **compatible** |
+
+
+
+### Option 2 - Api versioning
+
+Every time compatibility changes, API version is increased (keeping existing API version for backwards compatibility)
+
+So in `Example 1.` -> New field is added to central and completely new API version is added 
+In `Example 2`. -> Whole new API version is added with new logic keeping existing logic if possible, and keeping existing API
+
+*Pros:*
+- Allows for backwards compatibility
+
+*Cons:*
+- Could result in quite a bit of overhead and a lot of use cases to consider for backwards compatibility when `core` changes are made
+- Could result in different variations of end point versions and a bit of duplication to keep track of
+
+### Option 3 - Two way compatibility check
+
+Both central and remote keep track of `from` version of compatibility, and during sync these are exchanged and checked (either on remote or central, prefer remote to reduce logic slightly on central).
+
+Basicallly for `Example 1.`, when new field is added on remote, we add it to central, and change `from version of central` on remote to this never version of central. (previous version of remote will still be supported)
+
+For `Example 2.`, when api shape changes on central we update remote to use new api, and keep existing api support on remote, in this case `from verion of remote` will now be updated to match new `remote` version (this new remote version will still be backwards compatible with older central versions)
+
+An example:
+* 1.02 remote is compatible with `>=` 5.06 central
+* 1.02 remote is compatible with `>=` 5.07 central
+* 1.07 remote is compatible with `>=` 5.08 central (this is carried over from 1.02 verision of remote).
+* 5.08 is compatible with `>=` 1.02 of remote
+* 6 central is compatible with `>=` 1.02 of remote  (this is carried over from 5.07 verision of central).
+* 2 of remote is compatible with `>=` 6 of central
+
+So let's say 1.07 is trying to sync with 5.08, central says it's compatible with 1.02 and up and 1.07 knows it's compatible with 5.08 and up, thus it's ok to sync.
+
+2 is trying to sync with 5.08, central says it's compatible with 1.02 and up and remote knows it's compatible with 6 and up, sync is not allowed.
+
+1.02 is tyring to sync with 5.08. central says it's compatible with 1.02 and up and remote knows it's compatible with 5.06 and up, sync is not allowed
```

Comment:
🤔  in this example I think sync would be allowed. remote is compatible with 5.06 and up; and the central version is 5.08 which is 'up' from 5.06

Reply:
Oops, 🙏 for the catch
I think I was trying to show an example where remote thinks it's compatible but central disagrees, changed the example: 8cb9f5d31f7c318168ab26ccd639129ca4ee6343

and added headers: fa9904203bea2d7c604b7d7aaa562217b5e3fb21